.github/workflows/test.yaml

```diff
@@ -30,7 +30,8 @@ jobs:
       - run: docker buildx install
       - run: make setup-ci-image
       - run: ./bin/acorn install --image acorn:v-ci --skip-checks --acorn-dns=disabled --network-policies=true
-      - run: TEST_ACORN_CONTROLLER=external TEST_FLAGS="--junitfile test-summary.xml" make test
+      - run: TEST_FLAGS="--junitfile test-summary.xml" make unit
+      - run: TEST_ACORN_CONTROLLER=external TEST_FLAGS="--junitfile test-summary.xml" make integration
```

Comment:
What would your thoughts be on breaking out these two steps into their own jobs? I see a few benefits here:

1. Unit tests shouldn't (at least in theory) need an Acorn installation or image built. So the separated Unit job can ignore those parts.
2. The Unit and Integration tests can run in parallel which would speed up the feedback loop.
3. Each test type gets their own summary.

The main con is that we increase the amount of action minutes we use but not by that much.

Reply:
Totally agree that we should break this out into two distinct jobs. I chose this route for now because we effectively still get two separate `go test` runs with separate timeouts, but without the hassle of rewiring the test output/slack reporting.

I was thinking about doing that as a follow up. WDYT?

Reply:
Yep! Sounds good to me. Totally fine with this being a follow-up.